nucliadb_vectors/src/data_types/dtrie_ram.rs

```diff
@@ -71,127 +63,123 @@ impl<Prop> DTrie<Prop> {
             }
         }
     }
-    pub fn get(&self, key: &[u8]) -> Option<&Prop> {
-        match key {
-            [] => self.value.as_ref(),
-            [head, tail @ ..] => self
-                .go_table
-                .get(head)
-                .and_then(|n| n.get(tail))
-                .or(self.value.as_ref()),
-        }
-    }
-    pub fn iter(&self) -> DTrieIter<Prop> {
-        DTrieIter::new(self)
-    }
-    pub fn delete(&mut self, key: &[u8]) {
-        self.inner_delete(key);
-    }
-}
-
-pub struct DTrieIter<'a, V> {
-    stack: Vec<(Vec<u8>, &'a DTrie<V>)>,
-    crnt_key: Vec<u8>,
-    crnt_trie: &'a DTrie<V>,
-}
-impl<'a, V> DTrieIter<'a, V> {
-    fn new(trie: &'a DTrie<V>) -> DTrieIter<'a, V> {
-        DTrieIter {
-            stack: Vec::new(),
-            crnt_key: Vec::new(),
-            crnt_trie: trie,
-        }
+    pub fn get(&self, key: &[u8]) -> Option<SystemTime> {
+        self.inner_get(key, None)
     }
-}
-impl<'a, V> Iterator for DTrieIter<'a, V> {
-    type Item = (Vec<u8>, &'a V);
-    fn next(&mut self) -> Option<Self::Item> {
-        for (index, child) in self.crnt_trie.go_table.iter() {
-            let mut key = self.crnt_key.clone();
-            key.push(*index);
-            self.stack.push((key, child));
-        }
-        match self.stack.pop() {
-            None => None,
-            Some((new_key, new_trie)) => {
-                self.crnt_key = new_key;
-                self.crnt_trie = new_trie;
-                if !self.crnt_trie.is_value() {
-                    self.next()
-                } else {
-                    self.crnt_trie
-                        .value
-                        .as_ref()
-                        .map(|v| (self.crnt_key.clone(), v))
-                }
-            }
-        }
+    pub fn prune(&mut self, time: SystemTime) {
+        self.inner_prune(time);
```

Comment:
Is it still useful to have these both `inner_prune` and `prune` methods ?

Reply:
`inner_prune` returns a bool that will not make any sense to the caller, is used by `inner_prune` to trim the trie.